### ISSUE TITLE
Finalize Neighbor Tables

### ISSUE THREAD

**Issue by ctcncgr** [legumeinfo/ongenome_schema#9]
Are we going a table per dataset, strings or a massive 1:1 table?  I am going to implement and benchmark the multiple tables as a first pass as postgres can handle thousands of tables before there are issues, and these will only crop up if doing a very very large union, which we may do, but we still won't even have close to 1K tables.

**Comment by sdash-github** [comment]
From yesterday's meeting with Andrew, I recollect:
-- First priority is to try one profile neighbor table per dataset.  
-- Once you do that I will be able to incorporate that to my scripts

## Naming the table:

-- Please use **dataset.accession_no** as suffix with a constant prefix **'profileneigbors'**.
Example profileneigbors_cicar1, profileneigbors_phavu1

**Comment by ctcncgr** [comment]
hey sudhansu I'm slightly confused by these examples based on dataset.accession_no.

Would it be more like 'profileneighbors_cicar1.ann1', or 'profileneighbors_cicar.1'?

**Comment by sdash-github** [comment]
The dataset we already have is "cicar1", this is the chickpea atlas (Deshi).
The dataset we are going to load is "phavu1" (a bean atlas).
This is the dataset table accession_no field (dataset.accession_no).
We need not bring in annotation version here again as it is already associated with the dataset.

**Comment by ctcncgr** [comment]
Ok gotcha.  This is going to be the top priority for the time being.

**Comment by ctcncgr** [comment]
Just thought about this.  dataset.accession_no should not be null.  Currently we allow it to be null.  I will change this.  I will also make the uniqueness check against dataset.accession_no instead of dataset.shortname if this makes sense to the group.

**Comment by adf-ncgr** [comment]
agreed that if we will rely on accession_no for table naming it should not allow nulls. the alternative that I think was on the table when we met was to use the table PK to "uniqu-ify" the profileneighbors table names. There might be problems with this approach but if we want to allow for null accession_no for some reason we could reconsider it. One thing we should consider is that  table naming will almost certainly constrain the set of allowed characters in accession_no; so Connor's earlier suggestion to use datastore folder naming like "'profileneighbors_cicar1.ann1" 
would be problematic in this respect, although I like the principle of connecting our accession_no to the naming we are using in the datastore- we could consider adopting the "keys" for this purpose, though I am not sure that is a good idea.
